lib/elixir/lib/path.ex

```diff
@@ -466,7 +466,10 @@ defmodule Path do
   def join(left, right) do
     left    = IO.chardata_to_string(left)
     os_type = major_os_type()
-    do_join(left, right, os_type) |> remove_dirsep(os_type)
+    path = do_join(left, right, os_type)
+    if byte_size(path) > 1,
```

Comment:
I am not very confident with this byte_size call. Maybe we could fix this by moving remove_dirsep inside the `do_join` calls below and only call it when appropriate? Can you give this alternative solution a try to see if things are better or worse? Thank you!